pkg/querier/querier.go

```diff
@@ -83,11 +85,13 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.DurationVar(&cfg.QueryStoreAfter, queryStoreAfterFlag, 12*time.Hour, "The time after which a metric should be queried from storage and not just ingesters. 0 means all queries are sent to store. If this option is enabled, the time range of the query sent to the store-gateway will be manipulated to ensure the query end is not more recent than 'now - query-store-after'.")
 	f.BoolVar(&cfg.ShuffleShardingIngestersEnabled, "querier.shuffle-sharding-ingesters-enabled", true, fmt.Sprintf("Fetch in-memory series from the minimum set of required ingesters, selecting only ingesters which may have received series since -%s. If this setting is false or -%s is '0', queriers always query all ingesters (ingesters shuffle sharding on read path is disabled).", validation.QueryIngestersWithinFlag, validation.QueryIngestersWithinFlag))
 	f.BoolVar(&cfg.PreferStreamingChunks, "querier.prefer-streaming-chunks", false, "Request ingesters stream chunks. Ingesters will only respond with a stream of chunks if the target ingester supports this, and this preference will be ignored by ingesters that do not support this.")
+	f.BoolVar(&cfg.PreferStreamingChunksStoregateway, "querier.prefer-streaming-chunks-storegateway", false, "Request storegateway stream chunks. Storegateway will only respond with a stream of chunks if the target storegateway supports this, and this preference will be ignored by storegateway that do not support this.")
```

Comment:
[nit] We usually write it "store-gateway":
```suggestion
	f.BoolVar(&cfg.PreferStreamingChunksStoregateway, "querier.prefer-streaming-chunks-storegateway", false, "Request store-gateways stream chunks. Store-gateways will only respond with a stream of chunks if the target store-gateway supports this, and this preference will be ignored by store-gateways that do not support this.")
```

Reply:
(also applies below)